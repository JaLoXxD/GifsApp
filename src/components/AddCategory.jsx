import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
	const [inputVal, setInputVal] = useState("");

	const onInputChange = (e) => {
		setInputVal(e.target.value);
	};

	const onSubmit = (e) => {
		e.preventDefault();
		const newCategory = inputVal.trim();
		if (newCategory.length <= 1) return;
		//setCategories((categories) => [...categories, inputVal]);
		onNewCategory(newCategory);
		setInputVal("");
	};

	return (
		<form onSubmit={(e) => onSubmit(e)} aria-label="form" >
			<input
				type="text"
				placeholder="Search gifs..."
				value={inputVal}
				onChange={(e) => {
					onInputChange(e);
				}}
			/>
		</form>
	);
};

AddCategory.propTypes = {
	onNewCategory: PropTypes.func.isRequired,
};
