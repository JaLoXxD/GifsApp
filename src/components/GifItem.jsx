import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
	return (
		<>
			<img src={url.url} alt={title} />
			<h2>{title}</h2>
		</>
	);
};

GifItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
