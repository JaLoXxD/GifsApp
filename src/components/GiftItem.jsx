import React from "react";

export const GiftItem = ({title, url}) => {
	return (
		<>
			<img src={url.url} alt={title} />
			<h2>{title}</h2>
		</>
	);
};
