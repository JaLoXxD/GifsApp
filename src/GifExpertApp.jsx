import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(["Dragon Ball", "GOW"]);

	const onAddCategory = (newCategory) => {
		if (categories.includes(newCategory)) return;
		setCategories([newCategory, ...categories]);
	};
	return (
		<>
			<h1>Gift App</h1>
			<AddCategory
				onNewCategory={(val) => {
					onAddCategory(val);
				}}
			/>
			{/* <button onClick={onAddCategory}>Add</button> */}
			{categories.map((el) => {
				return <GifGrid key={el} category={el} />;
			})}
		</>
	);
};
