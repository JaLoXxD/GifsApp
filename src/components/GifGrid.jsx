import { GiftItem } from "./GiftItem";

import { useFetchGifts } from "../hooks/useFetchGifts";

export const GifGrid = ({ category }) => {
	const { gifts, isLoading } = useFetchGifts(category);

	return (
		<>
			<h3>{category}</h3>

			{isLoading && <h2>Loading...</h2>}

			<div className="gifsGrid">
				{gifts.map((gift) => {
					return (
						<div className="gifCard" key={gift.id}>
							<GiftItem {...gift} />
						</div>
					);
				})}
			</div>
		</>
	);
};
