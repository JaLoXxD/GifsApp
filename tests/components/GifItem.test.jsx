import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("tests on GifItem component", () => {
	const gif = {
		title: "test",
		url: { url: "https://www.asdf.com/index.js" },
	};
	test("should make match with the snapshot", () => {
		const { container } = render(<GifItem {...gif} />);
		expect(container).toMatchSnapshot();
	});

	test("should show the image with the URL and ALT text sended", () => {
		render(<GifItem {...gif} />);
		screen.debug();
		console.log(screen.getByRole("img").src);
		const { src, alt } = screen.getByRole("img");
		expect(src).toBe(gif.url.url);
		expect(alt).toBe(gif.title);
	});

	test("should show the title in a paragraph", () => {
		render(<GifItem {...gif} />);
		expect(screen.findByText(gif.title)).toBeTruthy();
	});
});
