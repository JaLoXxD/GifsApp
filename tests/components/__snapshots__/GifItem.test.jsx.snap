// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on GifItem component should make match with the snapshot 1`] = `
<div>
  <img
    alt="test"
    src="https://www.asdf.com/index.js"
  />
  <h2>
    test
  </h2>
</div>
`;
